### PR TITLE
 Fix Advanced Single-Sided Natural Ventilation

### LINKED ISSUE #6143: Index mismatch in two arrays causes crash in an example input file with single side window ventilation
Issue overview
--------------
CPValue array of MultizoneCPValueDataTemp is allocated from 1 to numWindDir. However, CPValue array of valsByFacade is allocated from 0 to numWindDir. If no adjustment of both arrays, index mismatch occurs to cause the program crash.
 
line8920 in AirflowNetworkBalanceManager
		for ( int i = 1; i <= numWindDir; ++i ) {
				MultizoneCPValueDataTemp( FacadeNum ).CPValue( i ) = valsByFacade[ FacadeNum-1 ][ i ];
			}

This only happens in Version 8.7. Version 8.6 is OK.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
This fixes the issues that broke the advanced single-sided natural ventilation feature. There still is no example file using the feature. I'll submit a separate issue for that. Fixes #6143 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the appropriate labels must be added to this PR to be consumed into the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)

### CODE REVIEW

**Comment by jasondegraw** [comment]
The ERR diffs are expected and due to a correction in this PR. In the course of developing the original fix, we discovered that the inclusion of any roof leakage would result in an incorrect assignment of wind pressure coefficient. We decided to add a roof wind pressure coefficient curve to address this problem, which may cause changes in results of any model using this feature. A related issue is the lack of an example file, which I'll put in a separate issue for.

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 1]
```diff
@@ -2274,47 +2274,42 @@ namespace AirflowNetworkBalanceManager {
...
-				{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << "AirflowNetwork Model:Wind Pressure Coefficients, "; }
-				{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << CurveManager::GetCurveName( index ) + ", "; }
-				for ( j = 0; j < 11; ++j ) {
-					StringOut = RoundSigDigits( CurveManager::CurveValue( index, dirs[ j ] ), 2 );
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << StringOut + ','; }
-				}
-				StringOut = RoundSigDigits( CurveManager::CurveValue( index, dirs[ 11 ] ), 2 );
-				gio::write( OutputFileInits, fmtA ) << StringOut;
-			}
+		int numWinDirs = 11;
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:2277]: `numWindirs`  :rofl: 

Sounds like my accent.

> "What's the num' windirs on that there house there?"

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 2]
```diff
@@ -2274,47 +2274,42 @@ namespace AirflowNetworkBalanceManager {
...
-				{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << "AirflowNetwork Model:Wind Pressure Coefficients, "; }
-				{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << MultizoneCPValueData( i ).Name + ", "; }
-				for ( j = 1; j <= AirflowNetworkSimu.NWind - 1; ++j ) {
-					StringOut = RoundSigDigits( MultizoneCPValueData( i ).CPValue( j ), 2 );
-					{ IOFlags flags; flags.ADVANCE( "No" ); gio::write( OutputFileInits, fmtA, flags ) << StringOut + ','; }
-				}
-				StringOut = RoundSigDigits( MultizoneCPValueData( i ).CPValue( AirflowNetworkSimu.NWind ), 2 );
-				gio::write( OutputFileInits, fmtA ) << StringOut;
+
+		// The old version used to write info with single-sided natural ventilation specific labeling, this version no longer does that.
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:2284]: Feels cleaner now

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 3]
```diff
@@ -5131,18 +5130,22 @@ namespace AirflowNetworkBalanceManager {
...
+			// The wind pressure coeffients are stored temporarily in the "valsByFacade" vector and then
+			// converted into a table near the end of this else. There will be at least seven profiles 
+			// (four sides plus one roof plus two for each pair of windows). The name is thus a little
+			// misleading, as it isn't really the values by facade once you get beyond the first five.
+			std::vector< std::vector< Real64 > > valsByFacade( 5 );
 			for ( FacadeNum = 0; FacadeNum < 4; ++FacadeNum ) {
 				valsByFacade[ FacadeNum ] = std::vector< Real64 >( 36 );
 			}
+			FacadeNum = 4;
+			valsByFacade[FacadeNum] = std::vector< Real64 >(12);
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:5148]: What is this square bracket voodoo?  :+1:

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 4]
```diff
@@ -5165,15 +5168,42 @@ namespace AirflowNetworkBalanceManager {
 					valsByFacade[ FacadeNum - 1 ][ WindDirNum - 1 ] = 0.6 * std::log( 1.248 - 0.703 * std::sin( IncRad / 2.0 ) - 1.175 * pow_2( std::sin( IncRad ) ) + 0.131 * pow_3( std::sin( 2.0 * IncRad * SideRatioFac ) ) + 0.769 * std::cos( IncRad / 2.0 ) + 0.07 * pow_2( SideRatioFac * std::sin( IncRad / 2.0 ) ) + 0.717 * pow_2( std::cos( IncRad / 2.0 ) ) );
 				} // End of wind direction loop
 			} // End of facade number loop
+			// Add a roof
+			FacadeNum = 5;
+			SR = min(max(SideRatio, 0.25), 1.0);
```
Comment (Myoldmopar) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:5173]: Some missing padding in the parentheses; if there is anything functional to change I'll update that as well.

File: src/EnergyPlus/DataAirflowNetwork.cc [hunk 5]
```diff
@@ -279,11 +274,6 @@ namespace DataAirflowNetwork {
 		MultizoneSurfaceCrackData.deallocate();
 		MultizoneSurfaceELAData.deallocate();
 		MultizoneExternalNodeData.deallocate();
-		MultizoneCPArrayData.deallocate();
-		MultizoneCPArrayDataSingleSided.deallocate();
-		MultizoneCPValueData.deallocate();
-		MultizoneCPValueDataTemp.deallocate();
-		MultizoneCPValueDataTempUnMod.deallocate();
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataAirflowNetwork.cc:286]: Yay!

**Review by Myoldmopar — APPROVED** [review]
Looks good to me

**Comment by Myoldmopar** [comment]
DElight file is unrelated to this work.  Merging this in.  Thanks for the updated fix sir @jasondegraw .

**Pull request merged**